snappy_pipeline/workflows/somatic_variant_signatures/__init__.py

```diff
@@ -211,38 +198,64 @@ def __init__(self, workflow, config, config_lookup_paths, config_paths, workdir)
+            else:
+                sub_workflow = "somatic_variant_filtration"
         self.register_sub_workflow(sub_workflow, config.path_somatic_variant, "somatic_variant")
         # Copy over "tools" setting from somatic_variant_calling/ngs_mapping if not set here
+
+        tools = set(self.w_config.step_config["ngs_mapping"].tools.dna)
         if not config.tools_ngs_mapping:
-            config.tools_ngs_mapping = self.w_config.step_config["ngs_mapping"].tools.dna
+            config.tools_ngs_mapping = tools
+        else:
+            config.tools_ngs_mapping = set(config.tools_ngs_mapping) & tools
+        assert len(config.tools_ngs_mapping) > 0, "No valid ngs mapping tool"
+
+        tools = set(self.w_config.step_config["somatic_variant_calling"].tools)
         if not config.tools_somatic_variant_calling:
-            config.tools_somatic_variant_calling = self.w_config.step_config[
-                "somatic_variant_calling"
-            ].tools
-        if not config.tools_somatic_variant_annotation:
-            config.tools_somatic_variant_annotation = self.w_config.step_config[
-                "somatic_variant_annotation"
-            ].tools
-        if config.is_filtered:
-            if len(self.w_config.step_config["somatic_variant_filtration"].filter_list) > 0:
-                config.filters = []
-                config.filtered_regions = []
-            else:
-                if not config.filters:
-                    config.filters = list(
-                        self.w_config.step_config["somatic_variant_filtration"].filter_sets.keys()
-                    )
-                    config.filters.append("no_filter")
-                if not config.filtered_regions:
-                    config.filtered_regions = list(
-                        self.w_config.step_config["somatic_variant_filtration"].exon_lists.keys()
-                    )
-                    config.filtered_regions.append("genome_wide")
-        self.w_config.step_config["somatic_variant_signatures"] = config
+            config.tools_somatic_variant_calling = tools
+        else:
+            config.tools_somatic_variant_calling = set(config.tools_somatic_variant_calling) & tools
+        assert len(config.tools_somatic_variant_calling) > 0, (
+            "No valid somatic variant calling tool"
+        )
+
+        if config.has_annotation:
+            tools = set(self.w_config.step_config["somatic_variant_annotation"].tools)
+            if not config.tools_somatic_variant_annotation:
+                config.tools_somatic_variant_annotation = tools
+            config.tools_somatic_variant_annotation = (
+                set(config.tools_somatic_variant_annotation) & tools
+            )
+            assert len(config.tools_somatic_variant_annotation) > 0, (
+                "No valid somatic variant annotation tool"
+            )
+
+        if config.filtration_schema == FiltrationSchema.sets:
```

Comment:
what if it is not `FiltrationSchema.sets`?

Reply:
Again, some comments needed:
This baroque monstrosity replaces un-configured tools in the config by their values taken from previous steps. In case of filter `sets`, the mutational signatures will be computed on variants passing all sets, unless the user specifically selects one.
When filtration is done using `list`, there is no need to to any of that, because the signatures are computed only on the output of the step.
The logic is the same for `tumor_mutational_burden`.